forte/data/data_store.py

```diff
@@ -226,6 +230,217 @@ def __init__(
         """
         self.__deletion_count: dict = {}
 
+    def __getstate__(self):
+        r"""
+        In serialization,
+            1) will serialize the annotation sorted list as a normal list;
+            2) will remove `_onto_file_path`, `_dynamically_add_type`,
+                `tid_ref_dict`, `tid_idx_dict` and `deletion_count` to save space.
+        """
```

Comment:
why don't we simply remove the `None`s here?

Reply:
Then we need to loop through all elements during both serialization and deserialization.

Reply:
I see, so you are saying that we are looping through the elements anyway during `deserialization` so that we'd better keep `serialization` simple

Reply:
yes that's right